modules/gwlb/variables.tf

```diff
@@ -80,21 +159,47 @@ variable "backends" {
       }
     }
   }
-  type = map(any)
+  nullable = false
+  type = map(object({
+    name = string
+    tunnel_interfaces = map(object({
+      identifier = number
+      port       = number
+      protocol   = optional(string, "VXLAN")
+      type       = string
+    }))
+  }))
+  validation { # protocol
+    condition = (var.backends == null ?
+      true : alltrue(flatten([for k, v in var.backends :
+    [for p, r in v.tunnel_interfaces : contains(["VXLAN"], r.protocol)]])))
+    error_message = "The `protocol` property can be only \"VXLAN\"."
+  }
+  validation { # type
+    condition = (var.backends == null ?
+      true : alltrue(flatten([for k, v in var.backends :
+    [for p, r in v.tunnel_interfaces : contains(["Internal", "External"], r.type)]])))
+    error_message = "The `type` property can be one of \"Internal\", \"External\"."
+  }
 }
 
 variable "lb_rule" {
   description = <<-EOF
-  Load balancing rule config. Available options:
-  - `name`              - (Optional|string) Name for the rule. Defaults to `var.frontend_ip_config.name`.
-  - `load_distribution` - (Optional|string) Refer to [provider docs](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lb_rule#load_distribution).
-  EOF
-  default     = null
-  type        = map(string)
-}
+  Load balancing rule configuration.
 
-variable "tags" {
-  description = "Azure tags to apply to the created resources."
+  Available options:
+  - `name`              - (`string`, optional) name for the rule.
+  - `load_distribution` - (`string`, optional, defaults to `Default`) specifies the load balancing distribution type
+                          to be used by the Gateway Load Balancer.
```

Comment:
I would add here a list of available options, or a link to external documentation (tf provider or azure) where that list can be found.

that's a general comment, for all variables add accept a finite list of arguments

Reply:
Done